### ISSUE TITLE
Strange memory error in gpu_memory.h

### ISSUE THREAD

**Issue by Student204161** [NVlabs/instant-ngp#1307]
Hi there,

At the moment, I get a strange memory error when trying to render a video & I think it might have something to do with instant-ngp trying to allocate already allocated memory.

Previously I got the same memory error when loading in my NeRF training dataset, and I fixed it by replacing some lines of code at line 452~ at dependencies/tiny-cuda-nn/include/tiny-cuda-nn/gpu_memory.h:
```
          // Reserve an address range that would be sufficient for housing the entire
      
          // available GPU RAM (if nothing else was using the GPU). This is unlikely     
      
          // to exhaust all available addresses (even if multiple GPUMemoryArenas are     
      
          // used simultaneously), while also ensuring that we never exhaust the     
      
          // reserved address range without running out of physical memory beforehand.    
       
          std::cerr << " gpu_memory:459 trying to reserve: " << m_max_size << "bytes" << std::endl;
      
          size_t padded_size;
      
          padded_size = (size_t)1*1024*1024*1024;
      
          CU_CHECK_THROW(cuMemAddressReserve(&m_base_address, padded_size, 0, 0, 0));

```
and further down the code at line 492 (that is after inserting he extra lines above),

```
                          size_t padded_size;
                          padded_size = (size_t)1*1024*1024*1024;
                          CU_CHECK_THROW(cuMemAddressFree(m_base_address, padded_size));

```
so as to use padded_size instead of m_max_size. This allows me to train & save models, export meshes and if I choose a low resolution (1300x900), low fps and a very short video duration I am able to render short video clips without instant-ngp crashing. If I don't, and I try to render, I get the following error:
`Attempting to free memory arena while it is still in use. 

10:36:28 ERROR    Uncaught exception: /zhome/a7/0/155527/Desktop/Bachelor-Project/instant-ngp/dependencies/tiny-cuda-nn/include/tiny-cuda-nn/gpu_memory.h:596 cuMemMap(m_base_address + m_size, n_bytes_to_allocate, 0, m_handles.back(), 0) failed with error CUDA_ERROR_INVALID_VALUE`

Which is weird for me since, I still have a lot of memory left on my GPU as seen below:

![billede](https://user-images.githubusercontent.com/97975723/229081902-f13a92b8-aa37-41f2-be69-128d87e2aabb.png)

From my understanding, instant-ngp only works with a single GPU so I have of course made sure to write
`export CUDA_VISIBLE_DEVICES=0`
in terminal.

Thanks for your attention.

**Comment by Tom94** [comment]
Hi, can you try latest master? `GPUMemoryArena` has a fallback mode for cards that don't support virtual memory mapping at all -- and it should work fine in your case as well. I changed the code so that it uses said fallback mode if the initial address reservation fails as seems to be the case for you.

**Comment by Student204161** [comment]
Hi Thomas,

Thanks for the quick response. I rebuilt newest master & the first thing I noticed was the NeRF I get in the display being quite grainy (even after finishing training):

![image](https://user-images.githubusercontent.com/97975723/229279695-8750d0d9-643c-4007-940e-1d67a1cb81fa.png)

However, I am able to render at 1920x1080 now, without crashing though. I think that the render quality is still the same if my eyes aren't fooling me:

For comparison, when I render at 1300x900 using new build I get:

https://user-images.githubusercontent.com/97975723/229282794-37b631a5-080a-43a0-acf3-14d85728a270.mp4

while my rendering in my old build with same settings:

https://user-images.githubusercontent.com/97975723/229283005-c9a4ed84-c963-469e-a7a9-d60598e7487b.mp4

The fallback mode seems to trigger:
![image](https://user-images.githubusercontent.com/97975723/229279638-3c307d78-30fa-4132-818f-224bf41b8a69.png)

In case the display being a bit pixelated becomes a problem for me to look at, my current solution is training all of my NeRFs using my old build and then when I want to render, it works if I copy my data to the new build and load in model and camera path & press render.

I appreciate the help.
